doc: PR and Issues templates + contributing guidelines

edkerk opened the pull request:
### Main improvements in this PR:
- doc: update PR and Issues templates with new branch names and reduce number of requested input to the most meaningful
- doc: update `CONTRIBUTING.md` to reflect recent changes

**I hereby confirm that I have:**

- [ ] Tested my code with [all requirements](https://github.com/SysBioChalmers/GECKO) for running GECKO
- [X] Selected `devel` as a target branch (top left drop-down menu)

mihai-sysbio commented on .github/CONTRIBUTING.md:
```diff
@@ -67,15 +67,17 @@ Here's how to set up GECKO for local development to contribute smaller features
-Finally, and for larger features that you want to work on collaboratively, you may consider to first request to join our development team to get write access to the repository so that you can create a branch directly in the main repository (or simply ask the administrator to create a branch for you). Once you have a new branch, you can push your changes directly to the main repository and when fi ...
+Try to keep pull requests relatively small: they should ideally provide one new feature or fix one bug. A few small bug-fixes maybe combined, but avoid packing pull requests with too many diverse and somewhat unrelated development. Smaller pull requests reduce reviewing time and make sure that code updates will make it into new releases faster.
+
+Finally, and for larger features that you want to work on collaboratively, you may consider to first request to join our development team to get write access to the repository so that you can create a branch directly in the main repository (or simply ask the administrator to create a branch for you). Once you have a new branch, you can push your changes directly to the main repository and when fi ...
```
```suggestion
Finally, and for larger features that you want to work on collaboratively, you may consider to first request to join our development team to get write access to the repository so that you can create a branch directly in the main repository. On this new branch, you can push your changes directly to the main repository and when finished, submit a pull request from that branch to `devel`. [See below](#development-team-guidelines) for more details.
```

mihai-sysbio commented on .github/CONTRIBUTING.md:
```diff
@@ -118,7 +120,7 @@ Follow all other steps in the same way. Also, when creating your pull request (o
 Every pull request must be approved by at least one reviewer before it can be merged. When reviewing someone else's pull request, keep in mind the following aspects:
 
-* **Compatibility:** Make sure the changes introduced in the PR do not generate errors when calling either `updateDatabases.m`, `enhanceGEM.m` or `generate_protModels.m`. To test the latter two functions, you may use [yeast 8.1.3](https://zenodo.org/record/1495482), as this will also help you validate any changes to the pipeline.
+* **Functionality:** The automated tests that are run as GitHub Actions should pass without any problems. Have you written a new function? Make sure that it is also tested in `tests/unittests/
```
```suggestion
* **Functionality:** The automated tests that are run as GitHub Actions should pass without any problems. Have you written a new function? Make sure that it is also tested in `tests/unittests/`.
```

edkerk replied:
```suggestion
* **Functionality:** The automated tests that are run as GitHub Actions should pass without any problems. Have you written a new function? Make sure that it is also tested in `tests/unit_tests/geckoCoreFunctionTests.m`.
```

mihai-sysbio commented on .github/CONTRIBUTING.md:
```diff
@@ -163,22 +165,12 @@ GECKO follows [semantic versioning](https://semver.org/):
+   - Specify the intended version in the title, e.g. `GECKO 3.0.1`
+   - Indicating all new features/fixes/etc. and referencing every previous pull request included (examples [here](https://github.com/SysBioChalmers/GECKO/releases)).
+   - If any [issue](https://github.com/SysBioChalmers/GECKO/issues) gets solved in the release, write in the pull request description "Closes #X", where "X" is the issue number. That way the issue will be automatically closed after merge.
+2. Wait at least a day + 1 approval + passing GitHub Actions.
```
```suggestion
2. Wait at least a day for at least one  approval. The GitHub Actions must also pass successfully.
```

mihai-sysbio commented on .github/pull_request_template.md:
```diff
@@ -1,8 +1,8 @@
-
+<!-- Pointwise mention what changes were made in what function. Examples: 
+- fix bug where `functionName` failed to do function X
+- updated all documentation -->
```
```suggestion
- updated all documentation 
- resolves issue #12345 -->
```

mihai-sysbio reviewed (approved):
Very nice additions!
Was there the intention to mention the GECKO 2 branch in the development guidelines? I remember suggesting it elsewhere, but now I'm not sure if it's a good idea.
Also, on a personal note, the name `devel` has been bugging me for some time. Everywhere I look on the web `develop` seems to be the branch name, so maybe we could go with that? That is - if we even want that branch at all. I think Johan made a comment on this topic recently. I see the arguments on both sides. What do you think @edkerk?

edkerk commented:
> Very nice additions! Was there the intention to mention the GECKO 2 branch in the development guidelines? I remember suggesting it elsewhere, but now I'm not sure if it's a good idea. Also, on a personal note, the name `devel` has been bugging me for some time. Everywhere I look on the web `develop` seems to be the branch name, so maybe we could go with that? That is - if we even want that branch at all. I think Johan made a comment on this topic recently. I see the arguments on both sides. What do you think @edkerk?

- I'd be happy to change to `develop`.
- No intention to meantion `gecko2`, any changes to that code are exceptional.
- Using a development branch was not so suitable with the recent high pace of development, but this will quite down now. One thing I did include was a mention that PRs should be "as small" as possible, which addresses some of the issues that Johan mentioned.

mihai-sysbio commented:
> * I'd be happy to change to `develop`.

That would be great! For the sake of small PRs, I imagine that could also be a separate PR, although the only renaming to be done that I know of is the _Contributing guidelines_ edited in the current PR.

edkerk commented:
Changed to `develop`, ready for merge.

mihai-sysbio commented on .github/CONTRIBUTING.md:
```diff
@@ -18,32 +18,32 @@ If you are unsure about the issue, consider asking first in our [Gitter chat roo
 * You checked that a similar issue does not exist already
 
-Feel free to also comment on any of the [open issues](https://github.com/SysBioChalmers/GECKO/issues). When doing so, please comply with our [code of conduct](https://github.com/SysBioChalmers/GECKO/blob/devel/.github/CODE_OF_CONDUCT.md).
+Feel free to also comment on any of the [open issues](https://github.com/SysBioChalmers/GECKO/issues). When doing so, please comply with our [code of conduct](https://github.com/SysBioChalmers/GECKO/blob/zzzzzzz/.github/CODE_OF_CONDUCT.md).
```
```suggestion
Feel free to also comment on any of the [open issues](https://github.com/SysBioChalmers/GECKO/issues). When doing so, please comply with our [code of conduct](https://github.com/SysBioChalmers/GECKO/blob/main/.github/CODE_OF_CONDUCT.md).
```
